### PR TITLE
DPL-1111-2: Refactor message management

### LINKED ISSUE #66: DPL-1111 Forward genome size from incoming messages to the Traction Service API
**User story**
The schema for messages coming from long read have the genome size for samples as part of the message.  Once tol-lab-share processes these and pushes the data to Traction, this field is missing from the message.  We should send the additional data.

**Who are the primary contacts for this story**
Stephen Inglis

**Who is the nominated tester for UAT**
James Watts

**Acceptance criteria**
To be considered successful the solution must allow:
- [x] Genome size received in messages from RabbitMQ is handled, validated and passed to Traction Service API.
- [x] Tests are updated to demonstrate that this process is working as intended.

**Dependencies**
This story is blocked by the following dependencies:
- sanger/traction-service/issues/1225 should be in place before deploying this change.

### PR DESCRIPTION
**Pull request by sdjmchattie**
Closes #66 

#### Changes proposed in this pull request

- Refactor messages to attempt to remove boilerplate and define better separation of message types.
- Move unit tests into test classes.
- Use built-in cache implementation for TaxonId MessageProperty.
- Use module `__init__.py` files to export classes from module.  This improves the number of imports needed.
- Remove tests that only test Python is working properly.
- Sort keys where they are hard to parse.
- Remove unused validation methods in MessageProperty class.

#### Instructions for Reviewers

_[All PRs] - Confirm PR template filled_  
_[Feature Branches] - Review code_

### CODE REVIEW

**Review by stevieing — APPROVED** [review]

File: setup.cfg [hunk 1]
```diff
@@ -4,6 +4,9 @@ max-complexity = 13
 statistics = True
 count = True
 show-source = True
+per-file-ignores =
+    # Ignore "imported but unused" errors in __init__.py files
+    __init__.py: F401
```
[the remark this answers is not in the document]
Comment (sdjmchattie) [reply setup.cfg]: Thanks.  I didn't know about the `__all__` variable.  It's a shame to have to declare boilerplate, but Copilot helps with that.

File: tol_lab_share/messages/properties/complex/scientific_name_from_taxon_id.py [hunk 2]
```diff
@@ -31,28 +41,17 @@ def value(self) -> Any:
...
+        except StatusCodeException as e:
+            self.trigger_error(
+                error_codes.ERROR_14_PROBLEM_ACCESSING_TAXON_ID,
+                text=f"HTTP code {e.status_code} - Problem when accessing taxon id {self._input.value}",
+            )
+            return None
 
     @staticmethod
     def reset_cache():
         """Static method to reset the cache shared by all instances"""
```
[the remark this answers is not in the document]
Comment (sdjmchattie) [reply tol_lab_share/messages/properties/complex/scientific_name_from_taxon_id.py:56]: This is still a static method though.  It's more of a convenience to be honest.  Someone could call the clear method directly if they wanted to.  I thought this made it clearer that the cache is associated with the usage of this class so I left it in.

[the remark this answers is not in the document]
Comment (sdjmchattie) [reply tol_lab_share/messages/properties/complex/scientific_name_from_taxon_id.py:56]: Maybe we're talking about different things.  The cache itself was being manually implemented (quite badly) here and I changed it to using the built-in `lru_cache` from Python's functools.  The new function will cache responses from the `get` requests up to 1000 of them, then it drops the least recently used one from the cache.

The method with this comment in, that you're referencing, is static though, as seen by the `@staticmethod` written above it.  This means you can call it without having an instance of the class, which makes sense since the cache function is outside this class and therefore shared across all instances.
